support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -65,45 +65,39 @@ object CreateZuoraSubscription {
   ): Future[HandlerResult[SendThankYouEmailState]] = {
     for {
       subscriptionData <- buildSubscriptionData(state, promotionService, GetCodeStatus.withDynamoLookup(redemptionService), today, config)
+          .withLogging("subscription data")
       subscribeItem = buildSubscribeItem(state, subscriptionData)
       identityId <- Future.fromTry(IdentityId(state.user.id))
         .withLogging("identity id")
       maybeDomainSubscription <- GetSubscriptionWithCurrentRequestId(zuoraService, state.requestId, identityId, state.product.billingPeriod, now)
         .withLogging("GetSubscriptionWithCurrentRequestId")
-      paymentMethodWithPaymentSchedule <-
+      paymentOrRedemptionData <-
         state.paymentMethod.leftMap(pm =>
           PreviewPaymentSchedule(subscribeItem, state.product.billingPeriod, zuoraService, checkSingleResponse)
-            .withLogging("PreviewPaymentSchedule").map(ps => (pm, ps))
+            .withLogging("PreviewPaymentSchedule").map(ps => PaymentMethodWithSchedule(pm, ps))
         ).leftSequence
-      thankYouState <- maybeDomainSubscription match {
-        case Some(domainSubscription) => skipSubscribe(state, requestInfo, paymentMethodWithPaymentSchedule, domainSubscription)
-          .withLogging("skipSubscribe")
-        case None => subscribe(subscribeItem, zuoraService).map(response => toHandlerResult(state, response, paymentMethodWithPaymentSchedule, requestInfo))
-          .withLogging("subscribe")
-      }
```

Comment:
I'm think there was a comment to extract this to a method in a previous PR, that PR was already big so I did it here.  It's only in this file so easy to revert if desired though.